test/tests/text_highlighting_test.py

```diff
@@ -78,6 +83,95 @@
+# |-mmm---| |--mmm--| |---mmm-| |-mmm---| |-mmmmm-| |-mmmmm-| |--mmm--| |--m-m--|
+# |----rrr| |--rrr--| |--rrr--| |--rrr--| |--rrr--| |--r-r--| |-rrrrr-| |-rrrrr-|
+# Collection choice is arbitrary.
+# Database choice is arbitrary.
+##############################################################################################
+case_morph_and_ruby = FakeEnvironmentParams(
+    config=config_big_japanese_collection,
+)
+CASE_MORPH_AND_RUBY_INPUT_TEXT = "12345  09876[def]  12345[abc]  12[abc] 34[abc]5  09876[def] 1 23[abc]45  012345[abc]  1234512345[abc]  0123[abc]45 1 23456[abc]  12345777[zyzzzzz]"
+CASE_MORPH_AND_RUBY_CORRECT_OUTPUT = '<span morph-status="unknown">12345</span> <ruby>09876<rt>def</rt></ruby> <span morph-status="unknown"><ruby>12345<rt>abc</rt></ruby></span> <span morph-status="unknown"><ruby>12<rt>abc</rt></ruby><ruby>34<rt>abc</rt></ruby>5</span> <ruby>09876<rt>def</rt></ruby> <span morph-status="unknown">1<ruby>23<rt>abc</rt></ruby>45</span> <ruby>0<span morph-status="unknown">12345</span><rt>abc</rt></ruby> <ruby><span morph-status="unknown">12345</span><span morph-status="unknown">12345</span><rt>abc</rt></ruby> <ruby>0<span morph-status="unknown">123</span><rt>abc</rt></ruby><span morph-status="unknown">45</span> <span morph-status="unknown">1</span><ruby><span morph-status="unknown">2345</span>6<rt>abc</rt></ruby> <ruby><span morph-status="unknown">12345</span><span morph-status="unknown">777</span><rt>zyzzzzz</rt></ruby>'
+
+case_morph_and_ruby_card_morphs = [
+    Morpheme(
+        lemma="12345",
+        inflection="12345",
+        highest_inflection_learning_interval=0,
+    ),
+    Morpheme(
+        lemma="777",
+        inflection="777",
+        highest_inflection_learning_interval=0,
+    ),
+]
+
+
+##############################################################################################
+#                                CASE: HIGHLIGHT BASED UNSPACED TEXT
+##############################################################################################
+# Highlight the text the best we can without spaces to determine where to place rubies.
+# Collection choice is arbitrary.
+# Database choice is arbitrary.
+##############################################################################################
+case_highlight_unspaced_params = FakeEnvironmentParams(
+    config=config_big_japanese_collection,
+)
+CASE_HIGHLIGHT_UNSPACED_INPUT_TEXT = "私は明日[あした]新しい[あたらしい]。"
+CASE_HIGHLIGHT_UNSPACED_OUTPUT = '<ruby><span morph-status="known">私</span>は<span morph-status="learning">明日</span><rt>あした</rt></ruby><ruby>新しい<rt>あたらしい</rt></ruby>。'
+case_highlight_unspaced_morphs = [
+    Morpheme(
+        lemma="私",
+        inflection="私",
+        highest_inflection_learning_interval=30,
+        highest_lemma_learning_interval=30,
+    ),
+    Morpheme(
+        lemma="明日",
+        inflection="明日",
+        highest_inflection_learning_interval=10,
+        highest_lemma_learning_interval=10,
+    ),
+]
```

Comment:
Made a whole new test using the minimal set of data that causes the issue. (I can understand your comment before about using more realistic data, so I'm all for putting this back as a full sentence if you like.

Reply:
I'll massage the tests a little bit, no worries.